File: src/main/kotlin/agendaonline/model/Agendamento_.java
package agendaonline.model;

import javax.persistence.metamodel.SingularAttribute;
import java.time.LocalDate;

public class Agendamento_ {

    public static volatile SingularAttribute<Agendamento, Long> codigo;
    public static volatile SingularAttribute<Agendamento, LocalDate> data_registro;
    public static volatile SingularAttribute<Agendamento, LocalDate> data_agendamento;
    public static volatile SingularAttribute<Agendamento, StatusAgendamento> StatusAgendamento;
    public static volatile SingularAttribute<Agendamento, Usuario_> usuario;
    public static volatile SingularAttribute<Agendamento, Cliente> cliente;

}

File: src/main/kotlin/agendaonline/model/Agendamento.kt
package agendaonline.model

import com.fasterxml.jackson.annotation.*
import java.time.LocalDate
import java.time.Month
import javax.persistence.*
import javax.validation.constraints.NotNull

@Entity
@JsonRootName("agendamento")
data class Agendamento (
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long = 0,
        @NotNull var data_registro: LocalDate? = null,
        @NotNull var data_agendamento: LocalDate? = null,
        @Enumerated(EnumType.STRING)
        @NotNull var status: StatusAgendamento? = null,
        @NotNull @OneToOne(fetch = FetchType.LAZY) @JoinColumn(name = "usuario_id")
        var usuario: Usuario = Usuario(),
        @NotNull @OneToOne(fetch = FetchType.LAZY) @JoinColumn(name = "cliente_id")
        var cliente: Cliente = Cliente()
)